### ISSUE TITLE
Untranslatable: Showing expired entries on database unlock

### ISSUE THREAD

**Issue by schlimmchen** [keepassxreboot/keepassxc#7787]
## Overview
The setting to select whether or not so show expired entries on database unlock or those which expire in some X amount of days is poorly translatable.

![Screenshot from 2022-04-04 07-57-06](https://user-images.githubusercontent.com/3578416/161482435-a52a035e-772d-4db8-af04-cdf9d015bc8f.png)
![Screenshot from 2022-04-04 08-03-14](https://user-images.githubusercontent.com/3578416/161483032-44f5c60f-896f-4c4f-ab02-49838d538957.png)

## Expected Design
Design that can be properly translated into foreign languages.

## Context
This already falls apart in German due to missing spaces in the translated strings. This could also be fixed in code, of course. Not mentioning that the actual translation is also broken **in more than one way** -- probably because of this very issue (poorly translatable, hard to understand the use and context for translators).

![Screenshot from 2022-04-04 07-58-42](https://user-images.githubusercontent.com/3578416/161482652-c7ab973d-ca77-4632-bbd9-b2c9206322d1.png)
![Screenshot from 2022-04-04 08-02-35](https://user-images.githubusercontent.com/3578416/161482972-221328f4-5d0a-448f-bbeb-6306c7ca6fbd.png)

Discussion about the original implementation in #4624.

## Transifex Issue

![Screenshot from 2022-04-04 08-00-54](https://user-images.githubusercontent.com/3578416/161482829-46b723a0-2231-4294-ad42-6eadf9a38206.png)

KeePassXC - 2.7.0
Operating System: all
Desktop Env: all
Windowing System: all

**Comment by schlimmchen** [comment]
I would like to argue that the first UI implementation by @ba32107 in https://github.com/keepassxreboot/keepassxc/issues/4624#issuecomment-1019161251 was much better with regard to being able to properly translate (replace "will expire in" with "expire within" and fix the plural):

![image](https://user-images.githubusercontent.com/3578416/161483747-8636e574-9770-4bd3-8865-f8948d208489.png)

Moreover, the plural is currently broken even in the English version (because "days" is a fixed suffix of the spinbox?), see the screenshot in the description where it ready "On database unlock, show entries that will expire in 1 days". Minor thing, but still. KeePassXC does care about plurals in general.

**Comment by phoerious** [comment]
> Minor thing, but still. KeePassXC does care about plurals in general.

We do. All plural strings have two (or more) alternative translations on Transifex. The single source string is indicated with a plural "(s)", the translation is expected to be the correct plural forms. Transifex lets you enter multiple translations for those strings based on the numerus. For spin boxes, that's sometimes not possible, because Qt only lets you define a fixed prefix or suffix. That's a Qt limitation.

**Comment by droidmonkey** [comment]
That whole feature needs to be rethought.

**Comment by schlimmchen** [comment]
So... A new issue titled "Rethink presentation of expired entries"? :wink:

May I suggest a "filter" that shows expired or about-to-be-expired entries? The same thing as `Expired` under `Database Tags` does but with a configurable timespan (making this phrasing issue still relevant) and moved to a section `Filters` (same level as `Database Tags`). The `Weak Passwords` pseudo-tag goes there (filters) as well. Just throwing out ideas...

**Comment by michaelk83** [comment]
There's no need for a new filter. Just make the `Expired` tag configurable (I think it already is?). If we use my banner suggestion from https://github.com/keepassxreboot/keepassxc/issues/7686#issuecomment-1079911012 , the banner can just link to the `Expired` tag. Then the settings become:
- `[✓] Remind of expired passwords every [time delta]`
- `[✓] Include passwords that will expire in [time delta]` (affects the `Expired` tag behavior)

`[time delta]` is the same widget from https://github.com/keepassxreboot/keepassxc/pull/7779#issuecomment-1087378467 . Though I guess it may be a bit overkill here.

And maybe rename the `Expired` tag to `Expiring`.

**Comment by droidmonkey** [comment]
We collapsed the two checkboxes into one purposefully. Don't want to go back to that again.

I want to extend the existing `is:expired` search entry by letting you add a `-##` to the end to indicate the days until expired offset. Example: `is:expired-3`. This is how I want to reimplement the expired on unlock logic, but it also needs an overhaul on presentation.

**Comment by michaelk83** [comment]
The new setting split is more about the reminder period ( #4167 ). But it also helps with the translations.

**Comment by schlimmchen** [comment]
> We collapsed the two checkboxes into one purposefully. Don't want to go back to that again.

I agree that it is elegant, but localization is important. IMHO the current solution is overly complicated to maintain. And simply not translatable to all languages.

> I want to extend the existing `is:expired` search entry by letting you add a `-##` to the end

Not that it matters much but I would support this only if search terms can be bookmarked to appear alongside `Expired` and `Weak Passwords` and if that feature (`is:expired-##`) was described in the help menu that appears when clicking to question mark on the right of the search bar.

> If we use my banner suggestion

Much better than switching the view to the `is:expired` search as it is much less obtrusive. IMHO we can then ditch the "Remind me every X of expiring passwords" and show the banner on every unlock. I also don't know how that would be (or currently is) implemented (Is the backoff-timer running only while KeePassXC is running? I don't the the database remembering when the last banner was shown).

**Comment by droidmonkey** [comment]
I am working on saved searches which are basically user customized tags at the top of that list. I think combining the banner display with the search improvements would be a win win. @michaelk83 can you create a feature request with your mock ups?

**Comment by michaelk83** [comment]
@droidmonkey , @schlimmchen , see #7791 . I've tweaked the wording a bit. Let me know if I missed anything.
